hack/loadtest/util/placement.go

```diff
@@ -79,148 +93,227 @@ func MeasureOnePlacement(ctx context.Context, hubClient client.Client, deadline,
 				// timeout
-				klog.V(3).Infof("the cluster resource placement `%s` delete timeout", crpName)
+				klog.V(3).Infof("the applied resources on cluster resource placement `%s` delete timeout", crpName)
 				LoadTestDeleteCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
 				deleteTimeoutCount.Inc()
-				break
+				return
 			}
-			continue
 		}
-		// check if the change is picked up by the member agent
-		var clusterWork workv1alpha1.Work
-		allRemoved := true
-		for _, clusterName := range clusterNames {
-			err := hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: fmt.Sprintf(utils.NamespaceNameFormat, clusterName)}, &clusterWork)
-			if err != nil || (err == nil && len(clusterWork.Status.ManifestConditions) != 2) {
-				klog.V(4).Infof("the resources `%s` in cluster namespace `%s` is not removed by the member agent yet", crpName, clusterName)
-				allRemoved = false
-				break
+	}
+}
+
+// check crp updated/completed before deletion
+func waitForCrpToComplete(ctx context.Context, hubClient client.Client, deadline, pollInterval time.Duration, deletionStartTime time.Time, crpName string, currency string, fleetSize string) {
+	startTime := time.Now()
+	applyDeadline := startTime.Add(deadline)
+	var crp v1beta1.ClusterResourcePlacement
+	var err error
+
+	// Create a timer that will fire when the deadline is reached
+	timer := time.NewTimer(deadline)
+
+	// Create a ticker that will fire at the specified poll interval
+	ticker := time.NewTicker(pollInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			// Context has been cancelled, exit the loop
+			return
+		case <-timer.C:
+			// Deadline has been reached, exit the loop
+			klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
+			LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
+			updateTimeoutCount.Inc()
+			return
+		case <-ticker.C:
+			// Poll for the CRP status
+			if err = hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: ""}, &crp); err != nil {
+				klog.ErrorS(err, "failed to get crp", "crp", crpName)
+				continue
+			}
+			// check if the condition is true
+			appliedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementAppliedConditionType))
+			synchronizedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementSynchronizedConditionType))
+			scheduledCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementScheduledConditionType))
+			if condition.IsConditionStatusTrue(appliedCond, 1) && condition.IsConditionStatusTrue(synchronizedCond, 1) && condition.IsConditionStatusTrue(scheduledCond, 1) {
+				// succeeded
+				klog.V(3).Infof("the cluster resource placement `%s` succeeded", crpName)
+				LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "succeed").Inc()
+				updateSuccessCount.Inc()
+				LoadTestUpdateLatencyMetric.WithLabelValues(currency, fleetSize).Observe(time.Since(deletionStartTime).Seconds())
+				timer.Stop()
```

Comment:
Hi Britania! Just a nit: these two Stop() calls should be in an outer defer statement I believe.